source/fundamentals/aggregation.txt

```diff
@@ -0,0 +1,198 @@
+
+Using ``find`` operations, you can:
+
+- select *which documents* to return
+- select *which fields* to return
+- sort the results
+
+Using ``aggregation`` operations, you can:
+
+- perform all ``find`` operations
+- rename fields
+- calculate fields
+- summarize data
+- group values
+
+The aggregation framework includes a wide array of computational and
+descriptive operators that allow you to produce computed results,
+whereas ``find`` operations provide a very limited number of data
+processing tools. 
+
+Note that aggregation operations have some :manual:`limitations </core/aggregation-pipeline-limits/>`:
+
+- Returned documents must not violate the :manual:`BSON-document size limit </reference/limits/#mongodb-limit-BSON-Document-Size>`
+  of 16 megabytes.
+
+- Pipeline stages have a memory limit of 100 megabytes by default. If necessary, you may exceed this limit by setting the ``allowDiskUse``
+  property of ``AggregateOptions`` to ``true``. See the
+  :node-api-4.0:`AggregateOptions API documentation <interfaces/aggregateoptions.html>`
+  for more details.
+
+.. important:: ``$graphLookup`` exception
+
+   The :manual:`$graphLookup
+   </reference/operator/aggregation/graphLookup/>` stage has a strict
+   memory limit of 100 megabytes and will ignore ``allowDiskUse``.
+
+Useful References
+~~~~~~~~~~~~~~~~~
+
+- :manual:`Aggregation pipeline </core/aggregation-pipeline/>`
+- :manual:`Aggregation stages </meta/aggregation-quick-reference/#stages>`
+- :manual:`Operator expressions </meta/aggregation-quick-reference/#operator-expressions>`
+
+Runnable Examples
+-----------------
+
+Base Setup
+~~~~~~~~~~
+
+Create a new Node.js file called ``AggTour.js`` and include the following statements:
+
+.. code-block:: javascript
+
+   const MongoClient = require('mongodb').MongoClient;
+   const assert = require('assert');
+
+Connect to a MongoDB Deployment
++++++++++++++++++++++++++++++++
+
+.. code-block:: javascript
+
+   // Connect to your MongoDB deployment.
+   const uri = 'mongodb://localhost:27017';
+
+   // Create or select an existing database.
+   const dbName = 'aggregationTest';
+
+   // Create a new MongoClient.
+   const client = new MongoClient(uri);
+
+   // Use connection method to connect to the Server.
+   client.connect(function(err, client) {
+        assert.equal(null, err);
+        console.log("Connected correctly to server");
+   });
+
+.. tip::
+
+   For information on connecting to your MongoDB deployment, see the :ref:`Connection Guide <connection>`.
+
+Insert Sample Data
+++++++++++++++++++
+
+The examples in this guide use a collection of documents corresponding
+to information about 10 dining establishments. Insert the sample data
+into the ``restaurants`` collection of the ``aggregationTest`` database:
+
+.. code-block:: javascript
+
+   // Select a database for use. 
+   const db = client.db(dbName);
+
+   // Select or create a collection.
+   const coll = db.collection("restaurants");
```

Comment:
Could these lines be moved to the earlier code block? Alternatively, I think it might make sense to remove the code about connecting to MongoDB and save that for another fundamentals page (Connection Guide Covers this in more depth).

Reply:
will restructure to link to the connection guide